src/Services/LocationService.php

```diff
@@ -0,0 +1,191 @@
+     * payload.
+     */
+    public function getAll($search = array(), $isAndCondition = true)
+    {
+        $sqlBindArray = array();
+
+        $sql = 'SELECT location.*, uuid_mapping.uuid FROM 
+                (SELECT
+                    uuid as target_uuid,
+                    CONCAT(fname,"\'s Home") as name,
+                    street,
+                    city,
+                    postal_code,
+                    state,
+                    country_code,
+                    phone_cell as phone,
+                    null as fax,
+                    null as website,
+                    email from patient_data
+                UNION SELECT
+                    uuid as target_uuid,
+                    name,
+                    street,
+                    city,
+                    postal_code,
+                    state,
+                    country_code,
+                    phone,
+                    fax,
+                    website,
+                    email from facility
+                UNION SELECT
+                    uuid as target_uuid,
+                    CONCAT(fname,"\'s Home") as name,
+                    street,
+                    city,
+                    zip as postal_code,
+                    state,
+                    null as country_code,
+                    phone,
+                    fax,
+                    url as website,
+                    email from users) as location
+                    LEFT JOIN uuid_mapping ON uuid_mapping.target_uuid=location.target_uuid';
+
+        if (!empty($search)) {
+            $sql .= ' WHERE ';
+            $whereClauses = array();
+            foreach ($search as $fieldName => $fieldValue) {
+                array_push($whereClauses, $fieldName . ' = ?');
+                array_push($sqlBindArray, $fieldValue);
+            }
+            $sqlCondition = ($isAndCondition == true) ? 'AND' : 'OR';
+            $sql .= implode(' ' . $sqlCondition . ' ', $whereClauses);
+        }
+
+        $statementResults = sqlStatement($sql, $sqlBindArray);
+
+        $processingResult = new ProcessingResult();
+        while ($row = sqlFetchArray($statementResults)) {
+            $row['uuid'] = UuidRegistry::uuidToString($row['uuid']);
+            $processingResult->addData($row);
+        }
+
+        return $processingResult;
+    }
+
+    /**
+     * Returns a single location record by id.
+     * @param $uuid - The location uuid identifier in string format.
+     * @return ProcessingResult which contains validation messages, internal error messages, and the data
+     * payload.
+     */
+    public function getOne($uuid)
+    {
+        $processingResult = new ProcessingResult();
+
+        $isValid = BaseValidator::validateIdMultiple(
+            "uuid",
+            array(
+                self::FACILITY_TABLE,
+                self::PRACTITIONER_TABLE,
+                self::PATIENT_TABLE
+            ),
+            $uuid,
+            true
+        );
```

Comment:
Is the validating the correct table(s). Aren't the uuid's that you are searching with stored in the uuid_mapping table?

Reply:
Yup, Also we don't need this Multiple Id function now :)